src/argilla/client/feedback/schemas.py

```diff
@@ -0,0 +1,157 @@
+)
+
+FETCHING_BATCH_SIZE = 250
+PUSHING_BATCH_SIZE = 32
+
+
+class ValueSchema(BaseModel):
+    value: Union[StrictStr, StrictInt]
+
+
+class ResponseSchema(BaseModel):
+    user_id: Optional[UUID] = None
+    values: Dict[str, ValueSchema]
+    status: Literal["submitted", "discarded"] = "submitted"
+
+    @validator("user_id", always=True)
+    def user_id_must_have_value(cls, v):
+        if not v:
+            warnings.warn(
+                "`user_id` not provided, so it will be set to `None`. Which is not an"
+                " issue, unless you're planning to log the response in Argilla, as "
+                " it will be automatically set to the active `user_id`.",
+                stacklevel=2,
+            )
+        return v
+
+
+class FeedbackRecord(BaseModel):
+    fields: Dict[str, str]
+    responses: Optional[Union[ResponseSchema, List[ResponseSchema]]] = None
+    external_id: Optional[str] = None
+
+    @validator("responses", always=True)
+    def responses_must_be_a_list(cls, v):
+        if not v:
+            return []
+        if isinstance(v, ResponseSchema):
+            return [v]
+        return v
+
+    class Config:
+        extra = Extra.ignore
+
+
+class FieldSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+class TextField(FieldSchema):
+    settings: Dict[str, Any] = Field({"type": "text"})
+    use_markdown: bool = False
+
+    @validator("use_markdown", always=True)
+    def update_settings_with_use_markdown(cls, v, values):
+        if v:
+            values["settings"]["use_markdown"] = v
+        return False
+
+
+FIELD_TYPE_TO_PYTHON_TYPE = {"text": str}
+
+
+class QuestionSchema(BaseModel):
+    name: str
+    title: Optional[str] = None
+    description: Optional[str] = None
+    required: Optional[bool] = True
+    settings: Dict[str, Any]
+
+    @validator("title", always=True)
+    def title_must_have_value(cls, v, values):
+        if not v:
+            return values["name"].capitalize()
+        return v
+
+
+# TODO(alvarobartt): add `TextResponse` and `RatingResponse` classes
+class TextQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "text", "use_markdown": False})
+    use_markdown: bool = False
+
+    @validator("use_markdown", always=True)
+    def update_settings_with_use_markdown(cls, v, values):
+        if v:
+            values["settings"]["use_markdown"] = v
+        return False
+
+    class Config:
+        validate_assignment = True
+        extra = Extra.forbid
+
+
+class RatingQuestion(QuestionSchema):
+    settings: Dict[str, Any] = Field({"type": "rating"})
+    values: List[int] = Field(unique_items=True)
+
+    @validator("values", always=True)
+    def update_settings_with_values(cls, v, values):
```

Comment:
not sure if in this case would be better to use a `pydantic.root_validator`, or use `@validator("settings")` and return the updated settings.